crates/cairo-lang-language-server/src/lib.rs

```diff
@@ -936,33 +936,48 @@ impl LanguageServer for Backend {
 
     #[tracing::instrument(level = "debug", skip_all, fields(uri = %params.text_document.uri))]
     async fn code_action(&self, params: CodeActionParams) -> LSPResult<Option<CodeActionResponse>> {
-        eprintln!("Code action");
         self.with_db(|db| {
             let mut actions = Vec::with_capacity(params.context.diagnostics.len());
             let file_id = file(db, params.text_document.uri.clone());
             let (node, _lookup_items) = get_node_and_lookup_items(db, file_id, params.range.start)?;
             for diagnostic in params.context.diagnostics.iter() {
-                let action = if let Some(NumberOrString::String(code)) = &diagnostic.code {
-                    match code.as_str() {
-                        codes::UNUSED_VARIABLE => unused_variable(
-                            db,
-                            &node,
-                            diagnostic.clone(),
-                            params.text_document.uri.clone(),
-                        ),
-                        _ => CodeAction::default(),
-                    }
-                } else {
-                    CodeAction::default()
-                };
-                actions.push(CodeActionOrCommand::from(action));
+                actions.extend(
+                    get_code_actions_for_diagnostic(db, &node, diagnostic, &params)
+                        .into_iter()
+                        .map(CodeActionOrCommand::from),
+                );
             }
             Some(actions)
         })
         .await
     }
 }
 
+fn get_code_actions_for_diagnostic(
+    db: &dyn SemanticGroup,
+    node: &SyntaxNode,
+    diagnostic: &Diagnostic,
+    params: &CodeActionParams,
+) -> Vec<CodeAction> {
+    let Some(code) = &diagnostic.code else {
+        debug!("diagnostic code is missing");
```

Comment:
```suggestion
        error!("diagnostic code is missing");
```

Reply:
I'd consider this as an error because we are the one sending the error codes with the diagnostic.